src/ipns/dns.rs

```diff
@@ -57,16 +59,18 @@ impl Future for DnsLinkFuture {
 pub async fn resolve(domain: &str) -> Result<IpfsPath, Error> {
     let mut dnslink = "_dnslink.".to_string();
     dnslink.push_str(domain);
-    let qname = Dname::from_str(&domain)?;
+    let qname = Dname::<Bytes>::from_str(&domain)?;
     let question = Question::new_in(qname, Rtype::Txt);
-    let query1 = StubResolver::new().query(question).compat();
+    let resolver = StubResolver::new();
+    let query1 = Box::pin(async move { resolver.query(question).await });
 
-    let qname = Dname::from_str(&dnslink)?;
+    let qname = Dname::<Bytes>::from_str(&dnslink)?;
     let question = Question::new_in(qname, Rtype::Txt);
-    let query2 = StubResolver::new().query(question).compat();
+    let resolver = StubResolver::new();
+    let query2 = Box::pin(async move { resolver.query(question).await });
 
     Ok(DnsLinkFuture {
-        query: select_ok(vec![query1, query2]),
+        query: select_ok(vec![query1 as FutureAnswer, query2]),
     }
```

Comment:
did you try either of the test_resolve1 or test_resolve2?

Reply:
In test_resolve2 we should just expect an IpfsPath and that's it. I did not know of the IpfsPath when writing that FIXME :)

Reply:
just tried it:
```
test ipns::dns::tests::test_resolve1 ... ok
test ipns::dns::tests::test_resolve2 ... FAILED

failures:

---- ipns::dns::tests::test_resolve2 stdout ----
thread 'ipns::dns::tests::test_resolve2' panicked at 'assertion failed: `(left == right)`
  left: `"/ipfs/bafybeidzfuvdzwkvfv4eonmbtpvkxkwxfxdfowww6wijxoj2chhryf7sc4"`,
 right: `"/ipfs/QmbV3st6TDZVocp4H2f4KE3tvLP1BEpeRHhZyFL9gD4Ut4"`', src/ipns/dns.rs:93:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    ipns::dns::tests::test_resolve2
```

Reply:
So I guess resolving works now :stuck_out_tongue:? I can update the path, but should I enable the tests? Probably not, as they could share the same fate as that one `kad` test we disabled today, no?

Reply:
Great that those work now as well! I this whole thing could be simplified a lot, but probably best to throw this in as is right away. No need to enable that test.. Need to learn if we can have tests enabled only when !ci.